commons/src/main/java/io/aiven/kafka/connect/common/source/input/AvroTransformer.java

```diff
@@ -56,7 +58,21 @@ public StreamSpliterator createSpliterator(final IOSupplier<InputStream> inputSt
 
             @Override
             protected void inputOpened(final InputStream input) throws IOException {
-                dataFileStream = new DataFileStream<>(input, datumReader);
+                InputStream decompressedInput = input;
+                if (!input.markSupported()) {
```

Comment:
are mark/reset generally not supported by the input stream, or what are the conditions potentially leading to this?